### PR TITLE
App

### PR DESCRIPTION
**Pull request by Nikita-Oskin**

### CODE REVIEW

File: streamlit_app.py [hunk 1]
```diff
@@ -1,38 +1,55 @@
...
+            image_data = uploaded_file.getvalue()
+            st.image(image_data)
+            return Image.open(io.BytesIO(image_data))
+        else:
+            return None
+
+    def print_predictions(pred):
+        classes = decode_predictions(pred, top=3)[0]
+        for cl in classes:
+            st.write(cl[1], cl[2])
```
Comment (SpirinEgor) [inline streamlit_app.py]: What is the reason you define these functions inside the `main` function? They don't use any scope information, so they can be defined as standalone functions.

**Review by SpirinEgor — CHANGES_REQUESTED** [review]

File: streamlit_app.py (hunk 1 above)
Comment (Nikita-Oskin) [reply streamlit_app.py]: done

File: streamlit_app.py [hunk 2]
```diff
@@ -1,38 +1,54 @@
...
+
+img = load_image()
+result = st.button("Recognize Image")
+if result:
+    num = preprocess_image(img)
+    pred = model.predict(num)
+    st.write("Recognition results:")
+    print_predictions(pred)
+else:
+    st.error("Choose an image!")
```
Comment (SpirinEgor) [inline streamlit_app.py]: And why do remove the main function now?) The idea was to split the functionality across multiple small functions, but right now you make your code global that reduce its flexibility and reusability

**Review by SpirinEgor — CHANGES_REQUESTED** [review]

File: streamlit_app.py (hunk 2 above)
Comment (Nikita-Oskin) [reply streamlit_app.py]: done

**Review by SpirinEgor — APPROVED** [review]

**Pull request merged**